browser/src/Services/InputManager.ts

```diff
@@ -0,0 +1,84 @@
+
+import { commandManager } from "./CommandManager"
+
+export type ActionFunction = () => boolean
+
+export type ActionOrCommand = string | ActionFunction
+
+export type FilterFunction = () => boolean
+
+export interface KeyBinding {
+    action: ActionOrCommand
+    filter?: FilterFunction
+}
+
+export interface KeyBindingMap {
+    [key: string]: KeyBinding[]
+}
+
+export class InputManager implements Oni.InputManager {
+
+    private _boundKeys: KeyBindingMap = {}
+
+    /**
+     * API Methods
+     */
+    public bind(keyChord: string, action: ActionOrCommand, filterFunction?: () => boolean) {
+        // tslint:disable-line no-empty-block
+
+        // TODO: add to existing binding
```

Comment:
What do you mean add to existing binding?
Are you saying if I bind c-j to next menu item also except c-n? perhaps have two methods, bind to override and bindMore to add them